src/basset.cpp

```diff
@@ -31,39 +31,40 @@ using std::literals::string_literals::operator""s;
 
 string json_escape(const string &str) {
   static const char hex[] = "0123456789abcdef";
+  char u[] = "u0000";
   std::string s;
 
-  for (const auto c : str) {
-    if (c == '\\' || c == '"') {
+  for (auto c : str) {
+    if (static_cast<unsigned char>(c) < 0x20) {
```

Comment:
Will it work if you simply compare to zero in addition to `0x20`?
```suggestion
    if (0 <= c && c < 0x20) {
```
It would be shorter, simpler and more "to the point" than `static_cast`.